pkg/utils/utils.go

```diff
@@ -48,14 +53,18 @@ var (
 	AllowedEventKindsMap map[EventKind]bool
 	// AllowedUpdateEventsMap is a map of resourceand namespace to updateconfig
 	AllowedUpdateEventsMap map[KindNS]config.UpdateSetting
-	// KubeClient is a global kubernetes client to communicate to apiserver
-	KubeClient kubernetes.Interface
-	// KubeInformerFactory is a global SharedInformerFactory object to watch resources
-	KubeInformerFactory informers.SharedInformerFactory
+	// DynamicKubeClient is a global dynamic kubernetes client to communicate to apiserver
+	DynamicKubeClient dynamic.Interface
+	// DynamicKubeInformerFactory is a global DynamicSharedInformerFactory object to watch resources
+	DynamicKubeInformerFactory dynamicinformer.DynamicSharedInformerFactory
+	// Mapper is a global DeferredDiscoveryRESTMapper object, which maps all resources present on
+	// the cluster, and create relation between GVR, and GVK
+	Mapper *restmapper.DeferredDiscoveryRESTMapper
 )
 
 // InitKubeClient creates K8s client from provided kubeconfig OR service account to interact with apiserver
 func InitKubeClient() {
+	var discoveryClient *discovery.DiscoveryClient
```

Comment:
@rahulchheda  We should use `DynamicKubeClient` which is globally declared. Otherwise e2e tests will fail as we are setting `DynamicKubeClient` with fake client but using discoveryClient instead.